src/v0.3/src/guest/call/syscall/nanosleep.rs

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::super::types::Argv;
+use super::Alloc;
+use crate::guest::alloc::{Allocator, Collect, Collector, Output};
+use crate::Result;
+
+use libc::{c_long, timespec};
+
+pub struct NanoSleep<'a> {
+    pub req: &'a mut timespec,
+}
+
+unsafe impl<'a> Alloc<'a> for NanoSleep<'a> {
+    const NUM: c_long = libc::SYS_clock_nanosleep;
+
+    type Argv = Argv<1>;
+    type Ret = ();
+
+    type Staged = Output<'a, timespec, &'a mut timespec>;
+    type Committed = Self::Staged;
+    type Collected = Result<()>;
+
+    fn stage(self, alloc: &mut impl Allocator) -> Result<(Self::Argv, Self::Staged)> {
+        let req = Output::stage(alloc, self.req)?;
+        Ok((Argv([self.req as _, req.offset()]), req))
```

Comment:
```suggestion
        Ok((Argv([req.offset(), rem_offset]), rem))
```

Where `rem_offset` if `NULL`, if `self.rem` is `None`, otherwise you should stage it as `Output` and use the `offset()` here